### PR TITLE
Server profile template idempotency

### LINKED ISSUE #101: Improve server profile template idempotency
### Scenario/Intent
On some usage scenarios, when running with the 'present' ensurable, the server profile template does not behave idempotently as expected and makes calls when it shouldn't. 

### Environment Details
*Puppet module for HPE OneView Version:* 2.0.0
*OneView SDK Version:* 3.1.0
*OneView Appliance Version:* 3.0
*OneView Client API Version:* 300
*Puppet Version:* 4.8
*Ruby Version:* 2.2.6
*Platform:* Windows

### Steps to Reproduce
Run complete scenarios with server profile template, specifying as many options as possible.

### Expected Result
If there are no changes between manifest runs, or if the change is something which should not matter, like order on a hash, then no create/update/patch calls should be made to OneView since it should be able to notice nothing has changed.

### Actual Result
On some scenarios it actually does not act as expected and runs update calls.

### PR DESCRIPTION
**Pull request by abiliogp**
### Description
ADD idempotency for Server Profile Template, it was tested in Synergy and C7000. 

### Issues Resolved
#101 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Comment by tmiotto** [comment]
Why are you changing server profile too?

It's a dependency from another PR? If so highlight it in the Description and link the PRs.

**Review by tmiotto — COMMENTED** [review]
I'm unsettled with the changelog... why is the v2.20 up in the changelog and the v3.2.0 down there, and why there is a separation of old logs between them if they are new additions...? :cold_sweat:

**Review by tmiotto — CHANGES_REQUESTED** [review]
I'm unsettled with the changelog... why is the v2.2.0 up in the changelog and the v3.2.0 down there, and why there is a separation of old logs between them if they are new additions...? :cold_sweat:

File: lib/puppet/provider/oneview_server_profile/c7000.rb [hunk 1]
```diff
@@ -24,14 +24,17 @@
...
+    super([:found, :get_available_targets, :get_available_networks, :get_available_servers,
+           :get_compliance_preview, :get_messages, :get_profile_ports, :get_transformation, :absent])
+
     # gets the connections' uris
     connections_parse if @data['connections']
     # gets the hash of filters for queries; in case it does not exist, query will be nil
     @query = @data.delete('query_parameters')
-    !@resourcetype.find_by(@client, @data).empty?
+
+    return @resourcetype.find_by(@client, @data).any? unless resource['ensure'] == :present
```
Comment (fgbulsoni) [inline lib/puppet/provider/oneview_server_profile/c7000.rb]: Do we really need this change? It seems like a copy from PR #94 and thus code duplication.
It is important to remember that the create method from oneview_resource actually does an update operation, and the way this is being written I'm not noticing any particular change from what is already done.

**Review by fgbulsoni — CHANGES_REQUESTED** [review]
For these changes to be acceptable, the scenarios in which the server profile template idempotency is failing would be a must to include in this PR and/or in the Issue related to it.
Also, would need confirmation that after these changes the 'idempotency' is working, while it was not before. 
Leaving request changes in the meanwhile.

**Pull request closed without merging**